src/main/scala/com/cognite/spark/datasource/CdpRelation.scala

```diff
@@ -65,9 +60,18 @@ abstract class CdpRelation[T <: Product: Decoder](config: RelationConfig, shortN
       },
       listUrl(),
       config,
-      pushdownFilters,
+      urlsWithFilters(filters, listUrl),
       cursors()
     )
+
+  def urlsWithFilters(filters: Array[Filter], uri: Uri): Seq[Uri] = {
+    val urlsWithFilters = fieldsWithPushdownFilter
```

Comment:
I think this might be better if written as
```
if (fieldsWithPushdownFilter.isEmpty) {
  Seq(uri)
} else {
  fieldsWithPushdownFilter.map(...).flatMap(...).distinct.map(...)
}

Reply:
Definitely, thanks.